src/lib.rs

```diff
@@ -50,29 +53,72 @@ pub struct CodecPipelineImpl {
     pub(crate) num_threads: usize,
 }
 
+#[gen_stub_pyclass_enum]
+enum StoreConfig {
+    Filesystem(FilesystemStoreConfig),
+    Http(HttpStoreConfig),
+    // TODO: Add support for more stores
+}
+
+impl<'py> FromPyObject<'py> for StoreConfig {
+    fn extract_bound(store: &Bound<'py, PyAny>) -> PyResult<Self> {
+        let name = store.get_type().name()?;
+        let name = name.to_str()?;
+        match name {
+            "LocalStore" => {
+                let root: String = store
+                    .getattr("root")?
+                    .call_method("as_posix", (), None)?
```

Comment:
maybe add a comment explaining that OpenDal expects POSIXy paths even on Windows (if that’s the case)